Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1751,19 +1751,19 @@ PhysicalParticleContainer::PushPX(WarpXParIter& pti,
     // This wraps the momentum and position advance so that inheritors can modify the call.
     auto& attribs = pti.GetAttribs();
     // Extract pointers to the different particle quantities
-    Real* AMREX_RESTRICT x = xp.dataPtr();
-    Real* AMREX_RESTRICT y = yp.dataPtr();
-    Real* AMREX_RESTRICT z = zp.dataPtr();
-    Real* AMREX_RESTRICT gi = giv.dataPtr();
-    Real* AMREX_RESTRICT ux = attribs[PIdx::ux].dataPtr();
-    Real* AMREX_RESTRICT uy = attribs[PIdx::uy].dataPtr();
-    Real* AMREX_RESTRICT uz = attribs[PIdx::uz].dataPtr();
-    Real* AMREX_RESTRICT Ex = attribs[PIdx::Ex].dataPtr();
-    Real* AMREX_RESTRICT Ey = attribs[PIdx::Ey].dataPtr();
-    Real* AMREX_RESTRICT Ez = attribs[PIdx::Ez].dataPtr();
-    Real* AMREX_RESTRICT Bx = attribs[PIdx::Bx].dataPtr();
-    Real* AMREX_RESTRICT By = attribs[PIdx::By].dataPtr();
-    Real* AMREX_RESTRICT Bz = attribs[PIdx::Bz].dataPtr();
+    Real* const AMREX_RESTRICT x = xp.dataPtr();
+    Real* const AMREX_RESTRICT y = yp.dataPtr();
+    Real* const AMREX_RESTRICT z = zp.dataPtr();
+    Real* const AMREX_RESTRICT gi = giv.dataPtr();
+    Real* const AMREX_RESTRICT ux = attribs[PIdx::ux].dataPtr();
+    Real* const AMREX_RESTRICT uy = attribs[PIdx::uy].dataPtr();
+    Real* const AMREX_RESTRICT uz = attribs[PIdx::uz].dataPtr();
+    const Real* AMREX_RESTRICT Ex = attribs[PIdx::Ex].dataPtr();
```

Comment:
Could we replace these lines by 
```
const Real * const AMREX_RESTRICT Ex = attribs[PIdx::Ex].dataPtr();
```
etc. to explicitly say that both the pointer and the pointed values are `const` in this scope?